spec/unit/recipes/source_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'spec_helper'
+
+describe 'prometheus::source' do
+  let(:chef_run) do
+    ChefSpec::SoloRunner.new(file_cache_path: '/var/chef/cache') do |node|
+      node.set['root_group'] = 'root'
+    end.converge(described_recipe)
+  end
+
+  it 'includes build-essential' do
+    expect(chef_run).to include_recipe('build-essential::default')
+  end
+
+  it 'creates a user with correct attributes' do
+    expect(chef_run).to create_user('prometheus').with(
+      system: true,
+      shell: '/bin/false',
+      home: '/opt/prometheus'
+    )
+  end
+
+  it 'creates a directory at /opt/prometheus' do
+    expect(chef_run).to create_directory('/opt/prometheus').with(
+      owner: 'root',
+      group: 'root',
+      mode: '0755',
+      recursive: true
+    )
+  end
+
+  %w( curl git-core mercurial gzip sed ).each do |pkg|
+    it 'installs #{pkg}' do
+      expect(chef_run).to install_package(pkg)
+    end
+  end
+
+  it 'checks out prometheus from github' do
+    expect(chef_run).to checkout_git("#{Chef::Config[:file_cache_path]}/prometheus-0.11.1").with(
```

Comment:
I could set ['prometheus']['source']['version'] to 0.11.1 at line 6? This way tests should work no matter what the actual version is. It could always just test with version 0.11.1